global_helpers/panther_greynoise_helpers.py

```diff
@@ -21,6 +22,15 @@ def __init__(self, sublevel):
             message = "Unknown Error Reading GreyNoise Data"
         super().__init__(message)
 
+class PantherIncorrectIPAddressMethodException(Exception):
+    def __init__(self, call_type):
+        if call_type is ImmutableCaseInsensitiveDict or call_type is str:
+            message = "~This is not the method you are looking for~ Try ip_address()"
+        elif call_type is ImmutableList or call_type is list:
+            message = "~This is not the method you are looking for~ Try ip_addresses()"
+        else:
+            message = f"Incorrect Method Exception, call_type: {call_type}"
+        super().__init__(message)
```

Comment:
I'm not particularly a huge fan of this as well as the addition of the `ip_addresses` function.

If we already know what function should be called then we shouldn't be telling the users to use that function. Instead, we should just return the results of the function that is supposed to be used

Reply:
I think it makes a lot of sense to have a separate function here for two reasons:
1. We can handle returning a list
2. We don't break or alter customer's existing detection rules.

I think the exception is actually a good thing, it returns an explicit failure and it is helpful by showing which method to use instead.

Reply:
@rleighton thoughts?

Reply:
If we have 1 function then callers will need to know when to expect a list and when to expect a scalar. A new function makes this clear. @wey-chiang Can you suggest a better way?

Reply:
I guess this should be fine if we're expecting users to only call the `context` function and the subsequent calls to `ip_address` and `ip_addresses` are made behind the scenes